### PR TITLE
Fix, improve Power Loss Recovery

### PR DESCRIPTION
**Pull request by Msq001**
### Description

1. The previous PLR used G92 Z to reset the Z axis positon to saved coordinate is not appropriate, because of G92  change the value of `position_shift[]` not `current_position[]`, when resume print after power loss, the `current_position[]` is 0 the `position_shift[Z_AXIS]` is saved z position,  and the `position_shift[]` will not be saved in PLR file, so when resume print again, the Z-axis coordinates will be confused.
In a word, PLR can only work normally once.
2. the `recovery.info` have not be cleared.  so when you stop current print, and print again.  the `info.current_position[Z_AXIS]` is the final Z-axis coordinate of the last printing, the `current_position[Z_AXIS] > info.current_position[Z_AXIS]` will be true until printing long time later, it will not saved printing status to PLR file from start print to z up to `info.current_position[Z_AXIS]`

### Benefits

1. fix.  use `G28 and G1 Z` instead of `G92 Z`, fix PLR can only work normally once.
2. fix.  fixed don't save printing status after stop print and print again
3. improve.  extrude some filament before resume print to clean nozzle

### CODE REVIEW

**Comment by thinkyhead** [comment]
`G92.0` was added specifically to set the position and not the offset. Are you saying it is broken?

**Comment by thinkyhead** [comment]
Hmm, I see that `G92.0` is not applicable. I must've gotten that mixed up with another version of Marlin I was working on.

The intent was to ensure that `current_position` is set to the correct native machine position which doesn't include the workspace offset. In my other SCARA-based Marlin project the `G92.0` command just directly sets the `current_position`. If I hadn't flubbed `G92.0` was different here, I might have added `G92.9` or just some custom G-code. (Custom G-codes are needed when we want some action to take place at the right point in the command flow.)

I will look over this PR and see if it accomplishes the intentions in the most economical manner.

File: Marlin/src/feature/power_loss_recovery.cpp [hunk 1]
```diff
@@ -315,15 +315,18 @@ void PrintJobRecovery::resume() {
     memcpy(&mixer.gradient, &info.gradient, sizeof(info.gradient));
   #endif
 
-  // Restore Z (plus raise) and E positions with G92.0
+  // home  
+  gcode.process_subcommands_now_P(PSTR("G28"));
```
Comment (thinkyhead) [inline Marlin/src/feature/power_loss_recovery.cpp]: Note: You can't home with `G28` when resuming a print. You will crush the object on the bed.

And, both Z-raise and homing are already done above. (Now fixed with `G92.9` in place of `G92.0`.)

**Comment by Msq001** [comment]
Current logic does not seem to apply to Delta machines

**Comment by thinkyhead** [comment]
Delta machines (`IS_KINEMATIC`) have been considered in some measure. `G28` homes to the top instead of to XY, for example. The Z raise before homing is redundant, but left in. If you have a delta for testing, please test and apply corrections for any bad behaviors observed.

**Pull request merged**